flexget/plugins/api/series_list.py

```diff
@@ -0,0 +1,341 @@
+series_list_parser.add_argument('name', help='Filter results by list name')
+
+
+@series_list_api.route('/')
+class SeriesListAPI(APIResource):
+    @api.response(200, model=return_lists_schema)
+    @api.doc(parser=series_list_parser)
+    def get(self, session=None):
+        """ Gets series lists """
+        args = series_list_parser.parse_args()
+        name = args.get('name')
+        series_lists = [series_list.to_dict() for series_list in sl.get_series_lists(name=name, session=session)]
+        return jsonify(series_lists)
+
+    @api.validate(new_list_schema)
+    @api.response(201, model=list_object_schema)
+    @api.response(500, description='List already exist', model=default_error_schema)
+    def post(self, session=None):
+        """ Create a new list """
+        data = request.json
+        name = data.get('name')
+        try:
+            series_list = sl.get_list_by_exact_name(name=name, session=session)
+        except NoResultFound:
+            series_list = None
+        if series_list:
+            return {'status': 'error',
+                    'message': "list with name '%s' already exists" % name}, 500
+        series_list = sl.SeriesListList(name=name)
+        session.add(series_list)
+        session.commit()
+        resp = jsonify(series_list.to_dict())
+        resp.status_code = 201
+        return resp
+
+
+@series_list_api.route('/<int:list_id>/')
+@api.doc(params={'list_id': 'ID of the list'})
+class SeriesListListAPI(APIResource):
+    @api.response(404, model=default_error_schema)
+    @api.response(200, model=list_object_schema)
+    def get(self, list_id, session=None):
+        """ Get list by ID """
+        try:
+            series_list = sl.get_list_by_id(list_id=list_id, session=session)
+        except NoResultFound:
+            return {'status': 'error',
+                    'message': 'list_id %d does not exist' % list_id}, 404
+        return jsonify(series_list.to_dict())
+
+    @api.response(200, model=empty_response)
+    @api.response(404, model=default_error_schema)
+    def delete(self, list_id, session=None):
+        """ Delete list by ID """
+        try:
+            series_list = sl.get_list_by_id(list_id=list_id, session=session)
+        except NoResultFound:
+            return {'status': 'error',
+                    'message': 'list_id %d does not exist' % list_id}, 404
+        session.delete(series_list)
+        return {}
+
+
+series_parser = api.parser()
+series_parser.add_argument('sort_by', choices=('title', 'added'), default='title',
+                           help='Sort by attribute')
+series_parser.add_argument('order', choices=('desc', 'asc'), default='desc', help='Sorting order')
+series_parser.add_argument('page', type=int, default=1, help='Page number')
+series_parser.add_argument('page_size', type=int, default=10, help='Number of series per page')
+
+series_identifiers_doc = "Use series identifier using the following format:\n[{'ID_NAME: 'ID_VALUE'}]." \
```

Comment:
This is executed when module is imported, it will make call to supported_ids .. which will get list of registered plugins, but plugins are not necessary yet loaded. Importing module should not have any side effects / dependencies.